### ISSUE TITLE
build issue: `double' is not a supported generic type argument, use `?' to box value types

### ISSUE THREAD

**Issue by alexmat** [dino/dino#1010]
building 73f012133ab7ba26e2279ddcb988873bdfd19663

```
# vala --version
Vala 0.51.3
```

running into the following issue:

```
[129/791] Generating C code for target QLITE_VALA_C                                                                                       
FAILED: qlite/src/database.c qlite/src/table.c qlite/src/column.c qlite/src/row.c qlite/src/statement_builder.c qlite/src/delete_builder.c qlite/src/insert_builder.c qlite/src/query_builder.c qlite/src/update_builder.c qlite/src/upsert_builder.c exports/qlite.vapi exports/qlite_internal.vapi exports/qlite.h exports/qlite_internal.h 
cd /tmp/pkgmk-dino/src/dino/build/qlite && /usr/bin/valac --color=always -C --header=/tmp/pkgmk-dino/src/dino/build/exports/qlite.h --internal-header=/tmp/pkgmk-dino/src/dino/build/exports/qlite_internal.h --vapi=/tmp/pkgmk-dino/src/dino/build/exports/qlite.vapi --internal-vapi=/tmp/pkgmk-dino/src/dino/build/exports/qlite_internal.vapi -b /tmp/pkgmk-dino/src/dino/qlite -d /tmp/pkgmk-dino/src/dino/build/qlite --pkg=gee-0.8 --pkg=glib-2.0 --pkg=gobject-2.0 --pkg=sqlite3 -g --target-glib=2.38 /tmp/pkgmk-dino/src/dino/qlite/src/database.vala /tmp/pkgmk-dino/src/dino/qlite/src/table.vala /tmp/pkgmk-dino/src/dino/qlite/src/column.vala /tmp/pkgmk-dino/src/dino/qlite/src/row.vala /tmp/pkgmk-dino/src/dino/qlite/src/statement_builder.vala /tmp/pkgmk-dino/src/dino/qlite/src/delete_builder.vala /tmp/pkgmk-dino/src/dino/qlite/src/insert_builder.vala /tmp/pkgmk-dino/src/dino/qlite/src/query_builder.vala /tmp/pkgmk-dino/src/dino/qlite/src/update_builder.vala /tmp/pkgmk-dino/src/dino/qlite/src/upsert_builder.vala
/tmp/pkgmk-dino/src/dino/qlite/src/column.vala:99.32-99.37: error: `double' is not a supported generic type argument, use `?' to box value types
    public class Real : Column<double> {
                               ^^^^^^
Compilation failed: 1 error(s), 0 warning(s)
```

**Comment by jasperla** [comment]
Same build error happens with the latest stable version of vala (0.50.4).

**Comment by licaon-kter** [comment]
seeing the same
Debian Sid, Vala 0.48.14

**Comment by mightyBroccoli** [comment]
Arch Linux has the same issue
Vala 0.50.4

**Comment by fiaxh** [comment]
This should be fixed by 9acb54df9254609f2fe4de83c9047d408412de28.

**Comment by jasperla** [comment]
@fiaxh thanks, I can confirm dino builds fine now.
